app/models/site.rb

```diff
@@ -207,6 +207,20 @@ def self.welcome_topic_banner_cache_key(user_id)
     "show_welcome_topic_banner:#{user_id}"
   end
 
+  def self.welcome_topic_exists_and_is_not_edited?
+    sql = <<~SQL
+      SELECT 1 FROM posts
+      JOIN topics
+      ON posts.topic_id = topics.id
+      WHERE posts.topic_id = #{SiteSetting.welcome_topic_id}
+      AND topics.deleted_at IS NULL
+      AND posts.post_number = 1
+      AND posts.version = 1
+      AND posts.created_at > '#{1.month.ago}';
+    SQL
+    DB.exec(sql) == 1
```

Comment:
I don't think dropping to raw SQL is necessary here. 

```suggestion
      Post.joins(:topic)
        .find_by(
          "topics.id = :topic_id AND topics.deleted_at IS NULL AND posts.post_number = 1 AND posts.version = 1 AND posts.created_at > :created_at",
          topic_id: SiteSetting.welcome_topic_id,
          created_at: 1.month.ago
        ).exists?
```